src/catalog/database_catalog.cpp

```diff
@@ -759,6 +749,8 @@ table_oid_t DatabaseCatalog::GetTableOid(transaction::TransactionContext *const
 
 bool DatabaseCatalog::SetTablePointer(transaction::TransactionContext *const txn, const table_oid_t table,
                                       const storage::SqlTable *const table_ptr) {
+  auto *tm = txn->GetTransactionManager();
+  txn->RegisterAbortAction([=]() { tm->DeferAction([=]() { delete table_ptr; }); });
```

Comment:
Can you add a comment explaining why we defer this action here, and why the catalog does it? I would expect the caller to check the return value of `SetTablePointer` and clean up the table it created if the call failed.